docker/run.sh

```diff
@@ -44,32 +45,40 @@ done
 
 echo_status "Starting up..."
 
-# Create Salt Config
-echo "file_client: local" > /etc/salt/minion
-echo "postgres.host: '${TETHYS_DB_HOST}'" >> /etc/salt/minion
-echo "postgres.port: '${TETHYS_DB_PORT}'" >> /etc/salt/minion
-echo "postgres.user: '${TETHYS_DB_USERNAME}'" >> /etc/salt/minion
-echo "postgres.pass: '${TETHYS_DB_PASSWORD}'" >> /etc/salt/minion
-echo "postgres.bins_dir: '${CONDA_HOME}/envs/${CONDA_ENV_NAME}/bin'" >> /etc/salt/minion
 
 if [[ $test = false ]]; then
   # Set extra ENVs
   export NGINX_USER=$(grep 'user .*;' /etc/nginx/nginx.conf | awk '{print $2}' | awk -F';' '{print $1}')
 
   # Apply States
   echo_status "Checking if DB is ready"
+  if [[ $db_engine == "django.db.backends.postgresql" ]]; then
+    # Create Salt Config for PostgreSQL
+    echo "postgres.host: '${TETHYS_DB_HOST}'" >> /etc/salt/minion
+    echo "postgres.port: '${TETHYS_DB_PORT}'" >> /etc/salt/minion
+    echo "postgres.user: '${TETHYS_DB_USERNAME}'" >> /etc/salt/minion
+    echo "postgres.pass: '${TETHYS_DB_PASSWORD}'" >> /etc/salt/minion
+    echo "postgres.bins_dir: '${CONDA_HOME}/envs/${CONDA_ENV_NAME}/bin'" >> /etc/salt/minion
+
+    db_check_count=0
+
+    until ${CONDA_HOME}/envs/${CONDA_ENV_NAME}/bin/pg_isready -h ${TETHYS_DB_HOST} -p ${TETHYS_DB_PORT} -U postgres; do
+      if [[ $db_check_count -gt $db_max_count ]]; then
+        >&2 echo "DB was not available in time - exiting"
+        exit 1
+      fi
+      >&2 echo "DB is unavailable - sleeping"
+      db_check_count=`expr $db_check_count + 1`
+      sleep 5
+    done
+  elif [[ $db_engine == "django.db.backends.sqlite3" ]]; then
+    echo_status "Using SQLite3 as the database"
+    # Add any SQLite-specific setup here if necessary
+  else
+    echo_status "Unsupported database engine: $db_engine"
```

Comment:
I think this is fine for now, but Tethys will support other engines. I use the Tethys docker with a pre-configured SQL Server DB.

Reply:
The comment regarding other databases not being supported has been removed.